---
output:
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
    fig_caption: yes
title: "MADE : Measuring Adaptive Distance and vaccine Efficacy using allelic barcodes"
params:
  id: "NA"
  def: "NA"
  strain: "NA"
  host: "NA"
  pass: "NA"
  post: "NA"
---

<span style="font-family:Arial; font-size:0.8em;">This report was generated using **MADE (Measuring Adaptive Distance and vaccine Efficacy using allelic barcodes)** at Genome Institute of Singapore, A*STAR, Singapore.</span>

<span style="font-family:Arial; font-size:0.8em;">In our companion study, we found that the strength of passage adaptation in embryonated eggs (measured as the Adaptive Distance or AD) negatively correlates with vaccine efficacy. Using this general principle, MADE enables users to calibrate the strength of egg passage adaptation and predict vaccine efficacy of a candidate influenza vaccine strain.</span>

<span style="font-family:Arial; font-size:0.8em;">For users preferring command line options, please refer to https://github.com/chenh1gis/MADE_docker_v1.</span>

<span style="font-family:Arial; font-size:0.8em;">For interactive interface, please refer to https://datahub.gis.a-star.edu.sg/made.</span>
  
### <span style="font-family:Arial; font-size:1em;">Virus Isolate Information</span>
```{r include = FALSE}
library(hash)
library(beeswarm)

id <- params$id
def <- params$def
strain <- params$strain
host <- params$host
pass <- params$pass
post <- params$post
date <- paste0(format(Sys.time(),tz="Asia/Singapore","%Y-%m-%d %H:%M:%S %Z"))

DAT=read.table("../data/H1N1pdm/H1N1pdm_enrichment_scores_327codons", header=TRUE, sep="\t")
ALLE=read.table("file_allele.txt",header=TRUE,sep="\t")
ALLE[,2]<-sapply(ALLE[,2],as.character) 
H=hash()
for (i in 1:nrow(ALLE))
  .set(H,keys=ALLE[i,1],values=ALLE[i,2])

ALLELE = data.frame(AminoAcid=character(),Codon=integer(),OR=double(),IsCandidate=integer(),stringsAsFactors = FALSE)
CODON=c(21,127,129,183,190,191,222,223,225)
for (i in 1:nrow(DAT))
  if (DAT[i,1]==21 | DAT[i,1]==127 | DAT[i,1]==129 | DAT[i,1]==183 | DAT[i,1]==190 | DAT[i,1]==191 | DAT[i,1]==222 | DAT[i,1]==223 | DAT[i,1]==225)
  {
    row=nrow(ALLELE)+1
    ALLELE[row,1]<-as.character(DAT[i,2])
    ALLELE[row,2]<-DAT[i,1]
    ALLELE[row,3]<-DAT[i,3]
    if (values(H,as.character(ALLELE[row,2]))==ALLELE[row,1])
    {
      ALLELE[row,4]<-1
    }else
    {
      ALLELE[row,4]<-0
    }
  }

GISAID=read.table("../data/H1N1pdm/H1N1pdm_background_strains_9alleles", header=TRUE, sep="\t")
GISAID[,c(1,3)] <- sapply(GISAID[,c(1,3)],as.character) 
GISAID[nrow(GISAID) + 1,1:3] = c("TEST","NA","TEST")

DAT[,c(2)] <- sapply(DAT[,c(2)],as.character) 
j=3
for(i in 1:9)
{
  j=j+1
  if (ALLE[which(ALLE$CODON==CODON[i]),2]!="NA")
  {
    if (DAT[which (DAT$Codon==CODON[i] & DAT$AminoAcid==ALLE[which(ALLE$CODON==CODON[i]),2]),3]!="NA")
    {GISAID[nrow(GISAID),j] = DAT[which(DAT$Codon==CODON[i] & DAT$AminoAcid==ALLE[which(ALLE$CODON==CODON[i]),2]),3]
    } else {stop(cat('The input of amino acid state over codon "',CODON[i],'" is not available in the list!\n', sep=''))}
  } else {stop(cat('The amino acid state over codon "',CODON[i],'" is missing in the input list!\n', sep=''))}
}
  
PCA <- princomp(GISAID[,4:12])
pc=c(1,2)
SCORE=PCA$scores
SCORE1=cbind(SCORE,GISAID$Passage)
SCORE=cbind(SCORE1,GISAID$Year)
colnames(SCORE)=c("PC1","PC2","PC3","PC4","PC5","PC6","PC7","PC8","PC9","PASSAGE","YEAR")

PC1test=PC2test=0
PC1bg=PC2bg=distT=0
TEST=SCORE[which(SCORE[,"PASSAGE"]=="TEST"),]
PC1test=as.numeric(TEST[1])
PC2test=as.numeric(TEST[2])
BACKGROUND=SCORE[which(SCORE[,"PASSAGE"]!="EGG_VACCINE" & SCORE[,"PASSAGE"]!="TEST"),]
for (j in 1:nrow(BACKGROUND))
  PC1bg=PC1bg+as.numeric(BACKGROUND[j,1])
  PC2bg=PC2bg+as.numeric(BACKGROUND[j,2])
PC1bg=PC1bg/nrow(BACKGROUND)
PC2bg=PC2bg/nrow(BACKGROUND)
disT=((PC1test - PC1bg)**2+(PC2test - PC2bg)**2)**0.5 # adaptive distance of tested isolate

AD = data.frame(Year=integer(),SpatialDistance=double())
for (i in 1:nrow(BACKGROUND))
{
  dis=((as.numeric(BACKGROUND[i,1]) - PC1bg)**2+(as.numeric(BACKGROUND[i,2]) - PC2bg)**2)**0.5
  AD[nrow(AD)+1,1:2] = c(BACKGROUND[i,11],dis)
}
AD=cbind(BACKGROUND[,"PASSAGE"],AD)
colnames(AD)=c("Passage","Year","SpatialDistance")
quantile=quantile(as.numeric(AD$SpatialDistance), c(.05, .95)) 
f=ecdf(as.numeric(AD$SpatialDistance))
percent=f(disT)

ADE=AD[which(AD[,"Passage"]=="EGG"),]
quantile1=quantile(as.numeric(ADE$SpatialDistance), c(.10, .90)) 
f=ecdf(as.numeric(ADE$SpatialDistance))
percentE=f(disT)

disT = trunc(disT*10^3)/10^3
quantile[[1]] = trunc(quantile[[1]]*10^3)/10^3
quantile[[2]] = trunc(quantile[[2]]*10^3)/10^3

# recommendation
recom <- paste('Please note that the 5% quantile of Adaptive Distance is ',quantile[[1]],'and the 95% quantile is ',quantile[[2]],'.')
```
  * <span style="font-family:Arial; font-size:0.8em;">Public_ID : `r id`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Definition : `r def`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Strain : `r strain`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Host : `r host`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Passage : `r pass`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Report Date : `r date`</span>

### <span style="font-family:Arial; font-size:1em;">Results & Recommendation</span>

<span style="font-family:Arial; font-size:0.9em;">**Probability of Egg Passage : **</span>
<span style="color:blue; font-family:Arial; font-size:0.9em;">**`r post`**</span>

<span style="font-family:Arial; font-size:0.9em;">**Adaptive Distance : **</span>
<span style="color:blue; font-family:Arial; font-size:0.9em;">**`r disT`**</span>

<span style="font-family:Arial; font-size:0.9em;">**Predicted Vaccine Efficacy : **</span>
<span style="color:blue; font-family:Arial; font-size:0.9em;">**`r veT`**</span>

<span style="font-family:Arial; font-size:0.9em;">*`r recom`*</span>

### <span style="font-family:Arial; font-size:1em;">Synopsis</span>

<span style="font-family:Arial; font-size:0.8em;">During vaccine production, influenza viruses are unavoidably propagated in embryonated eggs. In the culture expansion, flu viruses adapt to the egg environment, a process known as passage adaptation. In our companion study, we found that egg passage adaptation is driven by repeated substitutions (i.e. convergent evolution) over key codons and passage adaptation often leads to highly specific alleles in egg-passaged strains. Using a statistical analysis of these sites, we develop a metric of Adaptive Distance (AD) quantifying the strength of passage adaptation and show that there is a strong negative correlation between AD of a vaccine strain and vaccine efficacy. Based on these observations and principles, we developed MADE (Measuring Adaptive Distance and vaccine Efficacy using allelic barcodes) for vaccine developers to measure the strength of passage adaptation and predict the efficacy of a vaccine strain based on its nucleotide sequence.</span>

***

### <span style="font-family:Arial; font-size:1em;">Adaptive Distance</span>
```{r fig1, echo = FALSE, fig.align="center"}

############### PCA - all strains #################

minpc1=as.integer(min(as.integer(SCORE[,1]))-5)
minpc2=as.integer(min(as.integer(SCORE[,2]))-5)

num=array(0,dim=c(30,30))
for (i in 1:nrow(BACKGROUND))
  num[as.integer((as.numeric(BACKGROUND[i,1])-minpc1)/1),as.integer((as.numeric(BACKGROUND[i,2])-minpc2)/1)]=num[as.integer((as.numeric(BACKGROUND[i,1])-minpc1)/1),as.integer((as.numeric(BACKGROUND[i,2])-minpc2)/1)]+1

COUNT = data.frame(X=double(),Y=double(),Number=integer())
for (i in 1:30)
  for (j in 1:30)
    for (n in 1:num[i,j])
      COUNT[nrow(COUNT) + 1,1:3] = c(i*1+minpc1,j*1+minpc2,num[i,j])

smoothScatter(COUNT$X,COUNT$Y,colramp = colorRampPalette(c("cornflowerblue","slategray1","mistyrose", "pink","red")),pch="",xlab=paste0("PC", pc[1], " (", round(PCA$sdev[pc[1]]/sum(PCA$sdev)*100,0), "%)"), ylab=paste0("PC", pc[2], " (", round(PCA$sdev[pc[2]]/sum(PCA$sdev)*100,0), "%)"),cex.lab=1,cex.axis=.8,main="Principle component plot of all virus isolates")
points(x=PC1test,y=PC2test,pch=19,cex=1)
text(x=PC1test,y=PC2test-1,labels="vaccine strain", cex= 1, offset = 10)

percent = trunc(percent*10^3)/10^3
percentE = trunc(percentE*10^3)/10^3
percent = paste(round(100*percent, 2), "%", sep="")
percentE = paste(round(100*percentE, 2), "%", sep="")

# legend of fig 1

fig1l <- "The enrichment scores calculated at multiple codons will define a multidimensional space. Using principle component analysis, we projected the high dimensions down to the first two principle components (PC). When plotting all virus strains in the database onto this PCA map (each isolate is a dot ), we found that most of the virus strains in the PCA map are highly clustered and only a small proportion of strains passaged in embryonated eggs form separate clusters distinct from the major cluster. We thus defined an adaptive distance (AD) measuring the separation between the major cluster and the target strain."

fig1 <- paste('AD of the input strain is ',disT,'. This value is higher than ',percent,' of all strains in the database. If comparing to egg passage strains, this adaptive distance locates at the ',percentE,' quantile.',sep='')
```
  <span style="font-family:Arial; font-size:0.8em;">*`r fig1l`*</span>

  <span style="font-family:Arial; font-size:0.8em;">*`r fig1`*</span>
  
***

### <span style="font-family:Arial; font-size:1em;">Supporting Materials</span>
#### <span style="font-family:Arial; font-size:1em;">Approach</span>

<span style="font-family:Arial; font-size:0.8em;">MADE is a computational package developed for the research community to measure the strength of egg passage adaptation and predict the potential vaccine efficacy of a candidate strain based on its nucleotide sequence. Using the allelic state of the key codons driving egg passage adaptation (see our companion study and manual), MADE will:</span>

<span style="font-family:Arial; font-size:0.8em;">1) Measure the strength of egg passage adaptation in the input strain.</span>

<span style="font-family:Arial; font-size:0.8em;">2) compute the posterior probability that the input strain is passaged in embryonated eggs.</span>

<span style="font-family:Arial; font-size:0.8em;">3) predict the potential vaccine efficacy of the input strain.</span>

<span style="font-family:Arial; font-size:0.8em;">With all these functions, MADE allows users to calibrate the suitability and efficacy of a candidate strain for vaccine production.</span>

***
#### <span style="font-family:Arial; font-size:1em;">Enrichment scores</span>

```{r fig3, echo = FALSE, fig.align="center"}

##################### beeswarm #######################################

boxplot(OR ~ Codon, data = ALLELE,boxwex=.6, col="gray",ylab="Enrichment score",xlab="Codon position",cex.lab=1,cex.axis=0.8,main="Enrichment scores across key codons")
beeswarm=beeswarm(OR ~ Codon, data = ALLELE, col="black", pch=19,cex=.8,pwcol = 1 + as.numeric(IsCandidate),add=TRUE,position='center',corralWidth=0.2)

outlier <-0
OUTLIER=ALLELE[which(ALLELE[,"IsCandidate"]==1),]
for (i in 1:nrow(OUTLIER))
{
  if (OUTLIER[i,3]>=15)
  {
    outlier=outlier+1
    if (as.numeric(outlier)==1)
    {
      term=paste(OUTLIER[i,2],OUTLIER[i,1],sep='');
    }
    else
    {
      term=paste(term,' ',OUTLIER[i,2],OUTLIER[i,1],sep='');
    }
  }
}

# legend of fig 3

fig3l <- "Enrichment scores (ES) measures the enrichment of specific alleles in egg passaged strains (vs background sequences). High ES indicates strong egg passage adaptation."

if (as.numeric(outlier) != 1){
  fig3 <- paste('Allele ',term,' carries a high enrichment score in the focal strain',sep=''); 
} else {
  fig3 <- "No allele carries high enrichment score across those selected codons in the input strain.";
}
```

  <span style="font-family:Arial; font-size:0.8em;">*`r fig3l`*</span>

  <span style="font-family:Arial; font-size:0.8em;">*`r fig3`*</span>

***
#### <span style="font-family:Arial; font-size:1em;">Distribution of Egg Passaged Viruses</span>

```{r fig4, echo = FALSE, fig.align="center"}

############### PCA - egg strains #################

minpc1=as.integer(min(as.integer(SCORE[,1]))-5)
minpc2=as.integer(min(as.integer(SCORE[,2]))-5)

EGG=SCORE[which(SCORE[,"PASSAGE"]=="EGG"),]

num=array(0,dim=c(30,30))
for (i in 1:nrow(EGG))
  num[as.integer((as.numeric(EGG[i,1])-minpc1)/1),as.integer((as.numeric(EGG[i,2])-minpc2)/1)]=num[as.integer((as.numeric(EGG[i,1])-minpc1)/1),as.integer((as.numeric(EGG[i,2])-minpc2)/1)]+1

COUNT = data.frame(X=double(),Y=double(),Number=integer())
for (i in 1:30)
  for (j in 1:30)
    for (n in 1:num[i,j])
      COUNT[nrow(COUNT) + 1,1:3] = c(i*1+minpc1,j*1+minpc2,num[i,j])

smoothScatter(COUNT$X,COUNT$Y,colramp = colorRampPalette(c("cornflowerblue","slategray1","red")),pch="",xlab=paste0("PC", pc[1], " (", round(PCA$sdev[pc[1]]/sum(PCA$sdev)*100,0), "%)"), ylab=paste0("PC", pc[2], " (", round(PCA$sdev[pc[2]]/sum(PCA$sdev)*100,0), "%)"),cex.lab=1,cex.axis=.8,main="Principle component plot of egg passaged viral isolates")
points(x=PC1test,y=PC2test,pch=19,cex=1)
text(x=PC1test,y=PC2test-1,labels="vaccine strain", cex= 1, offset = 10)

# legend of fig 4

fig4l <- "Principle component plot of the enrichment scores across all egg passaged strains. Each strain is a dot in the PCA map. Egg-passaged strains seem to be clustered into different groups and bear different levels of egg passage adaptation."
```
  <span style="font-family:Arial; font-size:0.8em;">*`r fig4l`*</span>

***
#### <span style="font-family:Arial; font-size:1em;">Adaptive Distances of the Input Sequence</span>

```{r fig5, echo = FALSE, fig.align="center"}

############## Density ################

DISTANCE = data.frame(Year=integer(),Passage=character(),SpatialDistance=double(),stringsAsFactors = FALSE)
BACKGROUND=SCORE[(SCORE[,"PASSAGE"]=="EGG"),]
for (i in 1:nrow(BACKGROUND))
{
  PC1dis=as.numeric(BACKGROUND[i,1])
  PC2dis=as.numeric(BACKGROUND[i,2])
  dis=((PC1dis - PC1bg)**2+(PC2dis - PC2bg)**2)**0.5
  DISTANCE[nrow(DISTANCE) + 1,1:3] =   suppressWarnings(c(as.integer(BACKGROUND[i,11]),as.character(BACKGROUND[i,10]),dis))
}

d=density(as.numeric(DISTANCE$SpatialDistance))
plot(d, xlab="Adaptive distance",cex.lab=1,cex.axis=.8,main = "Adaptive distance of the input sequence against all egg passaged isolates")
polygon(d, col="gray", border="gray")
abline(v=disT,col='red',lwd=2) 

# legend of fig 5

fig5l <- "In general, adaptive distance (AD) calculated from egg passaged strains form two major clusters, indicating weak and strong signature of egg passage adaptation. The AD of the input sequence is indicated as vertical bar in the histogram."
```

  <span style="font-family:Arial; font-size:0.8em;">*`r fig5l`*</span>

***

#### <span style="font-family:Arial; font-size:1em;">Reference</span>
  * <span style="font-family:Arial; font-size:0.8em;">*Skowronski DM, et al. (2014) Low 2012-13 influenza vaccine effectiveness associated with mutation in the egg-adapted H3N2 vaccine strain not antigenic drift in circulating viruses. PloS one 9(3):e92153.*</span>
  * <span style="font-family:Arial; font-size:0.8em;">*Chen H, et al. (2016) Dynamic Convergent Evolution Drives the Passage Adaptation across 48 Years' History of H3N2 Influenza Evolution. Mol Biol Evol 33(12):3133-3143.*</span>
  * <span style="font-family:Arial; font-size:0.8em;">*Belongia EA, et al. (2016) Variable influenza vaccine effectiveness by subtype: a systematic review and meta-analysis of test-negative design studies. The Lancet. Infectious diseases 16(8):942-951.*</span>



