---
output:
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
    fig_caption: yes
title: "MADE : Measuring Adaptive Distance and vaccine Efficacy using allelic barcodes"
params:
  id: "NA"
  def: "NA"
  strain: "NA"
  host: "NA"
  pass: "NA"
  post: "NA"
---

<span style="font-family:Arial; font-size:0.8em;">This report was generated using **MADE (Measuring Adaptive Distance and vaccine Efficacy using allelic barcodes)** at Genome Institute of Singapore, A*STAR, Singapore.</span>

<span style="font-family:Arial; font-size:0.8em;">In our companion study, we found that the influenza viruses, especailly H3N2 subtype, undergo strong egg passage adaptation during egg vaccine production and the strength of passage adaptation (measured as the Adaptive Distance or AD) negatively correlates with vaccine efficacy. In terms of this principle, MADE enables users to examine the passage history and predict vaccine efficacy of a Candidate Influenza Vaccine (CVV).</span>

<span style="font-family:Arial; font-size:0.8em;">For users preferring command line options, please refer to https://github.com/chenh1gis/MADE_docker_v1.</span>

<span style="font-family:Arial; font-size:0.8em;">For interactive interface, please refer to http://39.105.1.41/made.</span>
  
### <span style="font-family:Arial; font-size:1em;">Virus Isolate Information</span>
```{r include = FALSE}
library(hash)
library(beeswarm)
library(knitr)
library(rpart)

id <- params$id
def <- params$def
strain <- params$strain
host <- params$host
pass <- params$pass
post <- params$post
date <- paste0(format(Sys.time(),tz="Asia/Singapore","%Y-%m-%d %H:%M:%S %Z"))

DAT=read.table("../data/H3N2/H3N2_enrichment_scores_329codons", header=TRUE, sep="\t")
ALLE=read.table("file_allele.txt",header=TRUE,sep="\t")
ALLE[,2]<-sapply(ALLE[,2],as.character) 
H=hash()
for (i in 1:nrow(ALLE))
  .set(H,keys=ALLE[i,1],values=ALLE[i,2])

ALLELE = data.frame(AminoAcid=character(),Codon=integer(),OR=double(),IsCandidate=integer(),stringsAsFactors = FALSE)
CODON=c(41,137,138,145,156,158,159,160,183,186,190,193,194,203,219,225,226,228,246,310)
for (i in 1:nrow(DAT))
  if (DAT[i,1]==41 | DAT[i,1]==137 | DAT[i,1]==138 | DAT[i,1]==145 | DAT[i,1]==156 | DAT[i,1]==158 | DAT[i,1]==159 | DAT[i,1]==160 | DAT[i,1]==183 | DAT[i,1]==186 | DAT[i,1]==190 | DAT[i,1]==193 | DAT[i,1]==194 | DAT[i,1]==203 | DAT[i,1]==219 | DAT[i,1]==225 | DAT[i,1]==226 |  DAT[i,1]==228 | DAT[i,1]==246 | DAT[i,1]==310)
  {
    row=nrow(ALLELE)+1
    ALLELE[row,1]<-as.character(DAT[i,2])
    ALLELE[row,2]<-DAT[i,1]
    ALLELE[row,3]<-DAT[i,3]
    if (values(H,as.character(ALLELE[row,2]))==ALLELE[row,1])
    {
      ALLELE[row,4]<-1
    }else
    {
      ALLELE[row,4]<-0
    }
  }

### examine the passage history

ALL <- read.table("../data/H3N2/H3N2_background_strains_20alleles_after2015", header=TRUE, sep="\t")
EGG <- ALL[which(ALL$Passage == "EGG"),]
NON <- ALL[which(ALL$Passage != "EGG"),]

#randomly sample the training sets from egges and nones, for adaboost algorithm
rset <- function(n,r=3) {#n is size of training set from egges, r is the 'ration' in my thesis,return the training set
  eggr <- sample(1:203,n,replace = F)
  nonr <- sample(1:30707,n*r,replace = F)
  eggtrain <- EGG[eggr,4:24]
  nontrain <- NON[nonr,4:24]
  train <- rbind(eggtrain,nontrain)
}

ab <- function(w,pd,ALL) {#function of Adaboost--focus on the 18 codons,w is the weight for each sample, pd is predict tag for each sample
  tagknow <- ALL['Tag']
  w1 <- w
  e <- 0#initialize error e
  n <- nrow(ALL)
  for(i in 1:n) {
    if(pd[i] != tagknow[i,1]) {
      e <- e+w1[i]#update the error e
    }
  }
  a <- (1/2)*log((1-e)/e)##weight of the current new weak classifier
  w2 = 1:n
  for(i in 1:n) {
    if(pd[i] != tagknow[i,1]) {
      b <- w1[i]*exp(a)
    }
    else{
      b <- w1[i]*exp(-a)
    }
    w2[i] <- b
  }
  n1 <- sum(w2)
  w3 = w2/n1#w3 is the new weight for each sample in next training turn
  o <- list(a,w3)
  return(o)
}

### undersampling algorithm  
### Function addt is to produce the weak predictors along with their weights.  
addt <- function(T,ALL){#T is the training times, to generate a strong classifier
  test <- ALL[,4:23]#test set is the whole samples with known tags
  n = nrow(ALL)
  w <- array()#weight for each sample
  for(i in 1:n) {
    w[i] = 1/n#initialize the w 
  }
  dt = list()#save each weak classifier in every traning turn, form them into a list
  par = array()#record each weak classifier's weight
  for(i in 1:T) {
    tr <- rset(180)#training set
    f <- Tag~X41+X137+X138+X145+X156+X158+X159+X160+X183+X186+X190+X193+X194+X203+X219+X225+X226+X228+X246+X310
    clf <- rpart(formula = f,data = tr,method = 'class')#generate a weak classifier based on the training set and decision tree method
    dt[[i]] <- clf#save it into the dt
    pd <- predict(clf,test)#predict results for test set of the current weak classifier clf
    pd1 <- array()
    for(j in 1:n) {
      if(pd[j,1] > pd[i,2]) {
        pd1[j] <- -1
      }
      else {
        pd1[j] <- 1#transform the pd into standard array of -1 or 1
      }
    }
    a <- ab(w, pd1,ALL)[[1]];w <- ab(w,pd1,ALL)[[2]]
    par[i] <- a
  }
  o <- list(dt,par)
  return(o)
}

dt = addt(100,ALL)[[1]];par = addt(100,ALL)[[2]]

#load('../data/H3N2/dt')#load 100 classifiers
#load('../data/H3N2/par')#load each classifier's weight
DAT1=read.table("../data/H3N2/H3N2_enrichment_scores_329codons_after2015", header=TRUE, sep="\t")
es <- data.frame(X41=numeric(),X137=numeric(),X138=numeric(),X145=numeric(),X156=numeric(),X158=numeric(),X159=numeric(),X160=numeric(),X183=numeric(),X186=numeric(), X190=numeric(),X193=numeric(),X194=numeric(),X203=numeric(),X219=numeric(),X225=numeric(),X226=numeric(),X228=numeric(),X246=numeric(),X310=numeric())
for(i in 1:20) {
  if (length(DAT1[which (DAT1$Codon==CODON[i] & DAT1$AminoAcid==ALLE[which(ALLE$CODON==CODON[i]),2]),3]) != "0" ) {
    es[1,i] = DAT1[which (DAT1$Codon==ALLE[i,1] & DAT1$AminoAcid==ALLE[i,2]),3]
  }else
  {es[1,i]=1} ### check the solution
}
s = 0
for(i in 1:length(dt)) {
  a = predict(dt[[i]],es)
  if(a[1,1]>a[1,2]){
  a1 = -1
  } else {
  a1 = 1
  }
  a2 <- a1*par[i]
  s = s+a2
}
  post <-sign(s)

  if (post==1) {
    post1 <- "yes"
  }else
  {
    post1 <- "no"
  }
### end

GISAID=read.table("../data/H3N2/H3N2_background_strains_20alleles", header=TRUE, sep="\t")
GISAID[,c(1,3)] <- sapply(GISAID[,c(1,3)],as.character) 
GISAID[nrow(GISAID) + 1,1:3] = c("TEST","NA","TEST")

DAT[,c(2)] <- sapply(DAT[,c(2)],as.character) 
CODON=c(41,137,138,145,156,158,159,160,183,186,190,193,194,203,219,225,226,228,246,310)
j=3
for(i in 1:20)
{
  j=j+1
  if (ALLE[which(ALLE$CODON==CODON[i]),2]!="NA")
  {
    if (length(DAT[which (DAT$Codon==CODON[i] & DAT$AminoAcid==ALLE[which(ALLE$CODON==CODON[i]),2]),3]) != "0" )
    {GISAID[nrow(GISAID),j] = DAT[which(DAT$Codon==CODON[i] & DAT$AminoAcid==ALLE[which(ALLE$CODON==CODON[i]),2]),3]
    } else {stop(cat('The input of amino acid state over codon "',CODON[i],'" is not available in the list!\n', sep=''))}
  } else {stop(cat('The amino acid state over codon "',CODON[i],'" is missing in the input list!\n', sep=''))}
}
  
PCA <- princomp(GISAID[,4:23])
pc=c(1,2)
SCORE=PCA$scores
SCORE1=cbind(SCORE,GISAID$Passage)
SCORE=cbind(SCORE1,as.character(GISAID$Year))
colnames(SCORE)=c("PC1","PC2","PC3","PC4","PC5","PC6","PC7","PC8","PC9","PC10","PC11","PC12","PC13","PC14","PC15","PC16","PC17","PC18","PC19","PC20","PASSAGE","YEAR")

DIS = data.frame(Year=integer(),SpatialDistance=double())
BACKGROUND=SCORE[which(SCORE[,"PASSAGE"]!="EGG_VACCINE" & SCORE[,"PASSAGE"]!="TEST"),]

PC1bg=PC2bg=0
for (j in 1:nrow(BACKGROUND))
{
  PC1bg=PC1bg+as.numeric(BACKGROUND[j,1])
  PC2bg=PC2bg+as.numeric(BACKGROUND[j,2])
}
PC1bg=PC1bg/nrow(BACKGROUND)
PC2bg=PC2bg/nrow(BACKGROUND)

for (i in 2010:2015)
{
  PC1vacc=PC2vacc=0

  dis=0
  VACCINE=SCORE[which(SCORE[,"YEAR"]==i & SCORE[,"PASSAGE"]=="EGG_VACCINE"),]
 
  for (j in 1:nrow(VACCINE))
  {
    PC1vacc=as.numeric(VACCINE[j,1])
    PC2vacc=as.numeric(VACCINE[j,2])
    dis=dis+((PC1vacc - PC1bg)**2+(PC2vacc - PC2bg)**2)**0.5
  }
  dis=dis/nrow(VACCINE)
  DIS[nrow(DIS) + 1,1:2] = c(i,dis)
}

PC1test=PC2test=0
PC1bg=PC2bg=distT=0
TEST=SCORE[which(SCORE[,"PASSAGE"]=="TEST"),]
PC1test=as.numeric(TEST[1])
PC2test=as.numeric(TEST[2])
BACKGROUND=SCORE[which(SCORE[,"PASSAGE"]!="EGG_VACCINE" & SCORE[,"PASSAGE"]!="TEST"),]
for (j in 1:nrow(BACKGROUND))
  PC1bg=PC1bg+as.numeric(BACKGROUND[j,1])
  PC2bg=PC2bg+as.numeric(BACKGROUND[j,2])
PC1bg=PC1bg/nrow(BACKGROUND)
PC2bg=PC2bg/nrow(BACKGROUND)
disT=((PC1test - PC1bg)**2+(PC2test - PC2bg)**2)**0.5 # adaptive distance of tested isolate

VE=c(0.46,0.32,0.40,0.10,0.07)
lm = lm(VE ~ DIS$SpatialDistance[1:5])
lmsum = summary(lm)
veT = lmsum$coefficients[2,1]*disT + lmsum$coefficients[1,1]  # vaccine efficacy of tested isolate

AD = data.frame(Year=integer(),SpatialDistance=double())
for (i in 1:nrow(BACKGROUND))
{
  dis=((as.numeric(BACKGROUND[i,1]) - PC1bg)**2+(as.numeric(BACKGROUND[i,2]) - PC2bg)**2)**0.5
  AD[nrow(AD)+1,1:2] = c(BACKGROUND[i,16],dis)
}
AD=cbind(BACKGROUND[,"PASSAGE"],AD)
colnames(AD)=c("Passage","Year","SpatialDistance")
quantile=quantile(as.numeric(AD$SpatialDistance), c(.05, .95)) 
f=ecdf(as.numeric(AD$SpatialDistance))
percent=f(disT)

ADE=AD[which(AD[,"Passage"]=="EGG"),]
quantile1=quantile(as.numeric(ADE$SpatialDistance), c(.10, .90)) 
f=ecdf(as.numeric(ADE$SpatialDistance))
percentE=f(disT)

disT = trunc(disT*10^3)/10^3
veT = trunc(veT*10^3)/10^3
quantile[[1]] = trunc(quantile[[1]]*10^3)/10^3
quantile[[2]] = trunc(quantile[[2]]*10^3)/10^3

# recommendation
recom <- paste('Please note that the 5% quantile of Adaptive Distance is ',quantile[[1]],'and the 95% quantile is ',quantile[[2]],'.')
```
  * <span style="font-family:Arial; font-size:0.8em;">Public_ID : `r id`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Definition : `r def`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Strain : `r strain`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Host : `r host`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Passage : `r pass`</span>
  * <span style="font-family:Arial; font-size:0.8em;">Report Date : `r date`</span>

### <span style="font-family:Arial; font-size:1em;">Results & Recommendation</span>

<span style="font-family:Arial; font-size:0.9em;">**Probability of Egg Passage : **</span>
<span style="color:blue; font-family:Arial; font-size:0.9em;">**`r post1`**</span>

<span style="font-family:Arial; font-size:0.9em;">**Adaptive Distance : **</span>
<span style="color:blue; font-family:Arial; font-size:0.9em;">**`r disT`**</span>

<span style="font-family:Arial; font-size:0.9em;">**Predicted Vaccine Efficacy : **</span>
<span style="color:blue; font-family:Arial; font-size:0.9em;">**`r veT`**</span>

<span style="font-family:Arial; font-size:0.9em;">*`r recom`*</span>

### <span style="font-family:Arial; font-size:1em;">Synopsis</span>

<span style="font-family:Arial; font-size:0.8em;">During traditonal egg vaccine production, influenza viruses are unavoidably propagated in embryonated eggs. And in this culture expansion, flu viruses adapt to the egg environment, a process known as passage adaptation. In our companion study, we found that egg passage adaptation is driven by repeated substitutions (i.e. convergent evolution) over a set of codons, which subsequently leads to highly enriched specific alleles in egg-passaged strains. Using a statistical analysis of these sites, we develop a metric of Adaptive Distance (AD) quantifying the strength of passage adaptation and show that there is a strong negative correlation between Adaptive Distance (AD) of a vaccine strain and Vaccine Efficacy (VE).</span>

<span style="font-family:Arial; font-size:0.8em;">Based on these observations and principles, we have developed a tool called MADE (Measuring Adaptive Distance and vaccine Efficacy using allelic barcodes). Through the application of a machine learning method, the passage history of Candidate Vaccine Viruses (CVVs) will be exaimined and simutaneously the adaptive Vaccine Efficacy (VEad) will be predicted. We hope that this tool will serve as a prescreening tool for WHO and vaccine developers with the aim to select the better vaccine strain devoid of egg passage adaptation and achieve a more optimistic vaccine performance.</span>

***

### <span style="font-family:Arial; font-size:1em;">Adaptive Distance</span>
```{r fig1, echo = FALSE, fig.align="center"}

############### PCA - all strains #################

minpc1=as.integer(min(as.integer(SCORE[,1]))-3)
minpc2=as.integer(min(as.integer(SCORE[,2]))-4)

num=array(0,dim=c(55,45))
for (i in 1:nrow(BACKGROUND))
  num[as.integer((as.numeric(BACKGROUND[i,1])-minpc1)/1),as.integer((as.numeric(BACKGROUND[i,2])-minpc2)/1)]=num[as.integer((as.numeric(BACKGROUND[i,1])-minpc1)/1),as.integer((as.numeric(BACKGROUND[i,2])-minpc2)/1)]+1

COUNT = data.frame(X=double(),Y=double(),Number=integer())
for (i in 1:55)
  for (j in 1:45)
    for (n in 1:num[i,j])
      COUNT[nrow(COUNT) + 1,1:3] = c(i*1+minpc1,j*1+minpc2,num[i,j])

smoothScatter(COUNT$X,COUNT$Y,colramp = colorRampPalette(c("cornflowerblue","slategray1","mistyrose", "pink","red")),pch="",xlab=paste0("PC", pc[1], " (", round(PCA$sdev[pc[1]]/sum(PCA$sdev)*100,0), "%)"), ylab=paste0("PC", pc[2], " (", round(PCA$sdev[pc[2]]/sum(PCA$sdev)*100,0), "%)"),cex.lab=1,cex.axis=.8,main="Principle component plot of all virus isolates")
points(x=PC1test,y=PC2test,pch=19,cex=1)
text(x=PC1test,y=PC2test-1,labels="vaccine strain", cex= 1, offset = 10)

percent = trunc(percent*10^3)/10^3
percentE = trunc(percentE*10^3)/10^3
percent = paste(round(100*percent, 2), "%", sep="")
percentE = paste(round(100*percentE, 2), "%", sep="")

# legend of fig 1

fig1l <- "The enrichment scores calculated at multiple codons will define a multidimensional space. Using principle component analysis, we projected the high dimensions down to the first two principle components (PC). When plotting all virus strains in the database onto this PCA map (each isolate is a dot ), we found that most of the virus strains in the PCA map are highly clustered and only a small proportion of strains passaged in embryonated eggs form separate clusters distinct from the major cluster. We thus defined an adaptive distance (AD) measuring the separation between the major cluster and the target strain."

fig1 <- paste('AD of the input strain is ',disT,'. This value is higher than ',percent,' of all strains in the database. If comparing to egg passage strains, this adaptive distance locates at the ',percentE,' quantile.',sep='')
```
  <span style="font-family:Arial; font-size:0.8em;">*`r fig1l`*</span>

  <span style="font-family:Arial; font-size:0.8em;">*`r fig1`*</span>
  
***

### <span style="font-family:Arial; font-size:1em;">Predicted Vaccine Efficacy</span>
```{r fig2, echo = FALSE, fig.align="center"}

############## scatterplot ###################

plot(DIS$SpatialDistance[1:5], VE, pch = 16, type = 'p', las = 1,cex=1,
     xlab = 'Adaptive distance',
     ylab = 'Vaccine efficacy',
     cex.lab=1,cex.axis=.8,
     main="Correlation between vaccine efficacy and adaptive distance"
     )
abline(lmsum$coefficients[1:2],lwd=2)
r2 = lmsum$adj.r.squared
mylabel = bquote(italic(R)^2 == .(format(r2, digits = 3)))
text(x = 28, y = 0.4, cex=1,labels = mylabel,font=4)
points(x=disT,y=veT,pch=19,cex=1,col="red")

post=1 ### need to update

if (disT>29)
{
  disTT=disT-1
}else if (disT<15)
{
  disTT=disT+1
  
}

if (post==1)
{
	veT1=formatC(veT,digits=2, format="f")
	veE=paste('VE = ',veT1,sep='')
	text(x=disTT,y=veT+0.1,labels=veE, cex=.8, offset = 10)
}
disT1=formatC(disT,digits=2, format="f") 
disE=paste('AD = ',disT1,sep='')


text(x=disTT,y=veT+0.07,labels=disE, cex=.8, offset = 10)
nx=c(0,disT)
ny=c(veT,veT)
lines(nx,ny,lwd=1.5,lty=2,col="red")
if (post==1)
{
	nx=c(disT,disT)
	ny=c(0,veT)
	lines(nx,ny,lwd=1.5,lty=2,col="red")
}
lmsum$coefficients[2,1] = trunc(lmsum$coefficients[2,1]*10^4)/10^4
lmsum$coefficients[1,1] = trunc(lmsum$coefficients[1,1]*10^4)/10^4

# legend of fig 2

fig2l="Using vaccine efficacy data (between year 2010-2015) curated by a recent study (Belongia EA, et al. 2016), we plotted the adaptive distance (AD) of past vaccine strains and the corresponding vaccine efficacy (VE). High R-square value of 0.729 indicates strong negative correlation between AD and VE."

fig2=paste('Using this linear relationship together with the estimated adaptive distance of ',disT,', the predicted vaccine efficacy for the input strain is ',veT,'.', sep='');
```
  <span style="font-family:Arial; font-size:0.8em;">*`r fig2l`*</span>

  <span style="font-family:Arial; font-size:0.8em;">*`r fig2`*</span>
  
***
### <span style="font-family:Arial; font-size:1em;">Supporting Materials</span>

#### <span style="font-family:Arial; font-size:1em;">Enrichment scores</span>

```{r fig3, echo = FALSE, fig.align="center"}

##################### beeswarm #######################################

boxplot(OR ~ Codon, data = ALLELE,boxwex=.6, col="gray",ylab="Enrichment score",xlab="Codon position",cex.lab=1,cex.axis=0.8,las=2, main="Enrichment scores across key codons")
beeswarm=beeswarm(OR ~ Codon, data = ALLELE, col="black", pch=19,cex=.8,pwcol = 1 + as.numeric(IsCandidate),add=TRUE,position='center',corralWidth=0.2)

outlier <-0
OUTLIER=ALLELE[which(ALLELE[,"IsCandidate"]==1),]
for (i in 1:nrow(OUTLIER))
{
  if (OUTLIER[i,3]>=15)
  {
    outlier=outlier+1
    if (as.numeric(outlier)==1)
    {
      term=paste(OUTLIER[i,2],OUTLIER[i,1],sep='');
    }
    else
    {
      term=paste(term,' ',OUTLIER[i,2],OUTLIER[i,1],sep='');
    }
  }
}

# legend of fig 3

fig3l <- "Enrichment scores (ES) measures the enrichment of specific alleles in egg passaged strains (vs background sequences). High ES indicates strong egg passage adaptation."

if (as.numeric(outlier) != 0){
  fig3 <- paste('Allele ',term,' carries a high enrichment score in the focal strain',sep=''); 
} else {
  fig3 <- "No allele carries high enrichment score across those selected codons in the input strain.";
}
```

  <span style="font-family:Arial; font-size:0.8em;">*`r fig3l`*</span>

  <span style="font-family:Arial; font-size:0.8em;">*`r fig3`*</span>

***
#### <span style="font-family:Arial; font-size:1em;">Adaptive Distances of the Input Sequence</span>

```{r fig5, echo = FALSE, fig.align="center"}

############## Density ################

DISTANCE = data.frame(Year=integer(),Passage=character(),SpatialDistance=double(),stringsAsFactors = FALSE)
BACKGROUND=SCORE[(SCORE[,"PASSAGE"]=="EGG"),]
for (i in 1:nrow(BACKGROUND))
{
  PC1dis=as.numeric(BACKGROUND[i,1])
  PC2dis=as.numeric(BACKGROUND[i,2])
  dis=((PC1dis - PC1bg)**2+(PC2dis - PC2bg)**2)**0.5
  DISTANCE[nrow(DISTANCE) + 1,1:3] =   suppressWarnings(c(as.integer(BACKGROUND[i,16]),as.character(BACKGROUND[i,15]),dis))
}

d=density(as.numeric(DISTANCE$SpatialDistance))
plot(d, xlab="Adaptive distance",cex.lab=1,cex.axis=.8,main = "Adaptive distance of the input sequence against all egg passaged isolates")
polygon(d, col="gray", border="gray")
abline(v=disT,col='red',lwd=2) 

# legend of fig 5

fig5l <- "In general, adaptive distance (AD) calculated from egg passaged strains form two major clusters, indicating weak and strong signature of egg passage adaptation. The AD of the input sequence is indicated as vertical bar in the histogram."
```

  <span style="font-family:Arial; font-size:0.8em;">*`r fig5l`*</span>

***

#### <span style="font-family:Arial; font-size:1em;">Reference</span>
  * <span style="font-family:Arial; font-size:0.8em;">*Skowronski DM, et al. (2014) Low 2012-13 influenza vaccine effectiveness associated with mutation in the egg-adapted H3N2 vaccine strain not antigenic drift in circulating viruses. PloS one 9(3):e92153.*</span>
  * <span style="font-family:Arial; font-size:0.8em;">*Chen H, et al. (2016) Dynamic Convergent Evolution Drives the Passage Adaptation across 48 Years' History of H3N2 Influenza Evolution. Mol Biol Evol 33(12):3133-3143.*</span>
  * <span style="font-family:Arial; font-size:0.8em;">*  Chen H, Alvarez JJS, Ng SH, Nielsen R, Zhai W. Passage adaptation correlates with the reduced efficacy of the influenza vaccine. Clin Infect Dis 2018.*</span>
  * <span style="font-family:Arial; font-size:0.8em;">*Belongia EA, et al. (2016) Variable influenza vaccine effectiveness by subtype: a systematic review and meta-analysis of test-negative design studies. The Lancet. Infectious diseases 16(8):942-951.*</span>



